Update UI for SD log period settings

mha1 opened the pull request:
fixes https://github.com/EdgeTX/edgetx/issues/2965

Changes:
- the log period range is now 0.1s to 25.5s
- upon creation of a new SD Logs SF the period is initialized to 1.0s
- text "Value" is changed to "Interval" to indicate parameter represents the log period
- text "Switch" is changed to "Trigger"

![image](https://user-images.githubusercontent.com/5615068/213929602-2684d198-88d4-4ce8-a6f1-a5f3c6ab41e8.png)

Eldenroot commented:
Please ping all translators

Eldenroot commented:
I am not at home right now, maybe @HThuren could you ping all of lang maintainers?

HThuren commented:
> I am not at home right now, maybe @HThuren could you please ping all of lang maintainers? Thank you

I ping them.

HThuren commented:
Hi translators,

@ParkerEde / @TheIsotopes DE 
@gagarinlg / @Eldenroot CZ
@ulfhedlund SE
@robustini IT
@zyren CN / TW
@ajjjjjjjj PL
@ToshihiroMakuuchi JP

Maybe you want to translate "Period" in your language file, please.

mha1 commented:
> Hi translators,
> 
> @ParkerEde / @TheIsotopes DE @gagarinlg / @Eldenroot CZ @ulfhedlund SE @robustini IT @zyren CN / TW @ajjjjjjjj PL @ToshihiroMakuuchi JP
> 
> Maybe you want to translate "Period" in your language file, please.

@HThuren they don't have the word. It's new with this PR

HThuren commented:
Translate -
#define TR_PERIOD                      "Period"

gagarinlg commented:
Wouldn't be "interval" a better choice than period?

HThuren commented:
> Wouldn't be "interval" a better choice than period?

New define ?
#define TR_INTERVAL "Interval"

mha1 commented:
Oxford dictionary definition of interval:
1.
an intervening time.
"after his departure, there was an interval of many years without any meetings"

2.
a pause or break in activity.
"an interval of mourning"

Oxford dictionary definition of period:
2.
PHYSICS
the interval of time between successive occurrences of the same state in an oscillatory or cyclic phenomenon, such as a mechanical vibration, an alternating current, a variable star, or an electromagnetic wave.

HThuren commented:
I think you decide, translate wanted for
#define TR_PERIOD "Period"

mha1 commented:
@HThuren Thanks for your support. How do you want to do this? Wait for the merge and then have individual PR's? Post translations here and me doing further commits?

HThuren commented:
> @HThuren Thanks for your support. How do you want to do this? Wait for the merge and then have individual PR's? Post translations here and me doing further commits?

When this only are one define, I think the simplest way are to have them here from the translators, and you merge in own branch. The way with PR in your fork/branch are - i think - too much in this case.
No long wait, translators are very active :-)

HThuren commented:
Hi translators, (maybe there where som confusion)

@ParkerEde / @TheIsotopes DE
@gagarinlg / @Eldenroot CZ
@ulfhedlund SE
@robustini IT
@zyren CN / TW
@ajjjjjjjj PL
@ToshihiroMakuuchi JP

Maybe you want to translate "Period" in your language file, please.
Provide in this thread your translation of:

**#define TR_PERIOD "Period"**

pfeerick commented:
When we can, we're trying to include translations as part of main PR. And just to add to the confusion on period vs interval :-P

IMO interval is indeed the more correct - period describes the *entire* time, rather than the time between samples. Hence the use of "period" in when defining interval ;)

![image](https://user-images.githubusercontent.com/5500713/213326972-cf2d201d-6784-4a91-b068-5e6db1aad54d.png)

![image](https://user-images.githubusercontent.com/5500713/213327362-6a91f975-2703-4aee-835a-0029a94b67de.png)

HThuren commented:
Why dont we make both defines, and the discussion will be which one to choose by programmer.
#define TR_PERIOD "Period"
#define TR_INTERVAL "Interval"

pfeerick commented:
Sounds good... the unused one now will probably get used in the future anyway.

HThuren commented:
DK translation
#define TR_PERIOD "periode"
#define TR_INTERVAL "interval"

mha1 commented:
@pfeerick We really should use frequency in Hz but we would end up with kinda small values :-).

https://www.mathsisfun.com/algebra/amplitude-period-frequency-phase-shift.html
https://en.m.wikipedia.org/wiki/Frequency

I'm ok with whatever you choose.

mha1 commented:
> DK translation
> #define TR_PERIOD "periode"
> #define TR_INTERVAL "interval"

@HThuren is it ok to start with a lower case letter?

HThuren commented:
> > DK translation
> > #define TR_PERIOD "periode"
> > #define TR_INTERVAL "interval"
> 
> @HThuren is it ok to start with a lower case letter?

arhg, no both must have capital first letter.
#define TR_PERIOD "Periode"
#define TR_INTERVAL "Interval"

ajjjjjjjj commented:
PL
```
#define TR_PERIOD "Okres"
#define TR_INTERVAL "Interwał"
```

MicheleVilla commented:
IT Italian translation

#define TR_PERIOD "Periodo"
#define TR_INTERVAL "Intervałlo"

TheIsotopes commented:
DE
```
#define TR_PERIOD "Zeitraum"
#define TR_INTERVAL "Intervall"
```

mha1 commented:
@ParkerEde @TheIsotopes 

I am very sure the German word for period is Periode.

Studyflix:
"Um die Frequenz zu berechnen, wird der Kehrwert der Periode T genutzt, also f = 1/T"

TheIsotopes commented:
@mha1 

imo `Zeitraum` is more comfortable than `Periode`. 

Just think of the women. ;-)

@ParkerEde What's your opinion?

robustini commented:
IT:

#define TR_PERIOD "Periodo"
#define TR_INTERVAL "Intervallo"

gagarinlg commented:
> @mha1
> 
> imo `Zeitraum` is more comfortable than `Periode`.
> 
> Floskel / Phrase: "In einem Zeitraum von 0,5 sekunden ..."
> 
> Just think of the women. ;-)
> 
> @ParkerEde What's your opinion?

"Zeitraum" and "Periode" have different meanings.  A sine wave has a "Periode", which is 1/frequency, "Zeitraum" would not be the right replacement.
"Zeitraum" defines a space in the time domain, where something will happen or is expected to happen, "Periode" is the time between reaching the same state again and again. 

Auf Deutsch:
Intervall: Abstand zwischen wiederkehrenden Ereignissen (in Zeit oder Raum)
Periode: Der Abstand, bei einem sich wiederholenden Systen, bei dem ein Zustand wieder erreicht wird. (in Zeit oder Raum)
Zeitraum: Ein Zeitbereich, in dem ein Ereignis passiert, oder erwartet wird

ParkerEde commented:
> @mha1
> 
> imo `Zeitraum` is more comfortable than `Periode`.
> 
> Floskel / Phrase: "In einem Zeitraum von 0,5 sekunden ..."
> 
> Just think of the women. ;-)
> 
> @ParkerEde What's your opinion?

I think "Periode" is more correct than "Zeitraum".

mha1 commented:
@gagarinlg @ParkerEde @TheIsotopes 
In the name of Math and Physics I hereby declare Periode to be the correct German word in the context of periodic events. 

This is PR is my most fun one yet.

gagarinlg commented:
> @gagarinlg @ParkerEde @TheIsotopes In the name of Math and Physics I hereby declare Periode to be the correct German word in the context of periodic events.
> 
> This PR is my most fun one yet.

I stand by my opinion that for the logging "interval" is the better choice, because the storing is non-continious.
;)

gagarinlg commented:
I will accept the majorities opionion, to not block a PR because of a minor wording issue, as this is not an international standard, but just firmware for RC transmitters.

ParkerEde commented:
"Intervall" is better than "Periode", "Periode" is better than "Zeitraum" is my opinion.

mha1 commented:
As decisions are well over my pay grade I'll happily accept management resolutions. Just spread the word.

pfeerick commented:
I think the majority consensus is that the final wording for the parameter will be `TR_INTERVAL` (i.e. time period between data captures) ... but we digressed slightly into whether for DE whether the translation for `TR_PERIOD` (i.e. time period for which data would be captured) should be "Periode" or "Zeitraum" ... But I also defer to the management/majority :grin:

ParkerEde commented:
DE:

#define TR_PERIOD "Periode"
#define TR_INTERVAL "Intervall"

mha1 commented:
@pfeerick Wait, aren't you the Senior Manager? No bailing on tough decisions. Pawn awaiting instructions.

pfeerick commented:
Well if you *want* me to put my foot down 😝  ... the thumbs of the native language DE seem to have settled on DE translation... and IMO it is TR_INTERVAL, not TR_PERIOD that applies with respect to the parameter for SD Log...

gagarinlg commented on radio/src/translations/cn.h:
```diff
@@ -935,6 +935,7 @@
 #define TR_TELEMETRY_TYPE              TR("类型", "回传类型")
 #define TR_TELEMETRY_SENSORS           "传感器"
 #define TR_VALUE                       "数值"
+#define TR_PERIOD                      "Period"
```
Interval is mising

gagarinlg commented on radio/src/translations/cz.h:
```diff
@@ -952,6 +952,7 @@
 #define TR_TELEMETRY_TYPE              TR("Typ", "Typ telemetrie")
 #define TR_TELEMETRY_SENSORS           "Senzory"
 #define TR_VALUE                       "Hodnota"
+#define TR_PERIOD                      "Period"
```
Interval is missing

mha1 replied:
Oops, resolved, see commit

gagarinlg reviewed (commented):
TR_INTERVAL is mssing for some languages, I did not tag all of them

robustini commented:
> Following a recent senior management decision I adapted the appearance of the SD Logs SF setup screen and also incorporated provided translations for EN, DA, DE, PL, IT. Please review your respective language files.
> 
> ![image](https://user-images.githubusercontent.com/5615068/213414693-ebed6f37-1462-48a4-ba2e-5a91e7879cfc.png)

For me is ok (IT)!

HThuren commented:
Danish DA is OK

mha1 commented:
fixed the problem where intervals < 1s would be displayed with two leading zeros.

before (2.8.0):
![image](https://user-images.githubusercontent.com/5615068/213441399-af1985c3-0c5d-46f0-8bdc-ad98855e6477.png)

after:
![image](https://user-images.githubusercontent.com/5615068/213441222-c09d360d-ec31-4f13-86c9-6ffaddce444e.png)

phileville commented:
Now I know why it was just "value" before :)

mha1 commented:
@pfeerick @gagarinlg 

@ParkerEde / @TheIsotopes DE
@gagarinlg / @Eldenroot CZ
@ulfhedlund SE
@robustini IT
@zyren CN / TW
@ajjjjjjjj PL
@ToshihiroMakuuchi JP

While tidying up the SD Logs screen I also noticed TR_FUNC uses abbreviated 4 character words for some languages including English ("Func"), while other languages have have full words up to 8 characters (e.g. ES, DE, FR). The limiting factor for length is NV14 with 12 characters possible. 

![image](https://user-images.githubusercontent.com/5615068/213536921-8d14b934-e9d3-4b09-b5c7-578e1b9a4599.png)

I hope management allows asking translators to check/update TR_FUNC. I have updated en.h to 
#define TR_FUNC "Function"

![image](https://user-images.githubusercontent.com/5615068/213538193-04f225e2-725b-4ed4-a0c4-a55e0bca5d92.png)

Logical Switches is the other screen benefiting:

![image](https://user-images.githubusercontent.com/5615068/213539544-ac7848bc-150f-4799-85e5-d9dd0e629510.png)

ajjjjjjjj commented:
PL
```
#define TR_FUNC                "Funkcja"
```

Eldenroot commented:
CZ:

`#define TR_FUNC                        "Funkce"`

Eldenroot commented:
Why do we have different UI style (padding) for these two screens?
![image](https://user-images.githubusercontent.com/778890/213548461-81a71d0b-8348-45c9-8220-61307836b471.png)

mha1 commented:
@Eldenroot  This is current main code, I haven't changed padding. Maybe the Logical Switches screen has not been migrated to lvgl yet.

gagarinlg commented:
Theogical switches page is still using the old UI code

ToshihiroMakuuchi commented:
JP:
#define TR_PERIOD "ピリオド"
#define TR_INTERVAL "インターバル"

#define TR_FUNC "ファンクション"

HThuren commented:
More DA
#define TR_FUNC                        TR("Funk", "Funktion")

But maybe also change TR_SWITCH to specific one for special functions, then not only switch are the event. fx
EN: #define TR_SF_SWITCH              TR("Event", "Switch7Event")

DA: #define TR_SF_SWITCH              TR("Udløser", "Kontakt/udløser")

robustini commented:
IT
#define TR_FUNC TR("Funz", "Funzione")

pfeerick commented:
I would be more inclined to go the way of "Trigger" for what is being described as `TR_SF_SWITCH` ... thoughts??

mha1 commented:
@HThuren

General question to TR_FUNC: I checked and found STR_FUNC being used in 3 places:
- gui/colorlcd Logical Switches
- gui/colorlcd Special Functions
- gui/128x64  Logical Swiitches

I guess the 128x64 case forces TR_FUNC to have a compatibility definition TR(x,y) if strings longer than n characters are used for colorlcd. What is the maximum number of characters a string for x can have?
TR_FUNC has no compatibility definitions so far but has longer definitions for some languages, e.g, ES, FR, PT. That's why I believe those languages having longer strings already need to be changed to compatibility definitions depending on the answer to the above question about maximum number of characters in the compatibility case.

Switch definitions for Special Functions
If there is consensus I can do this. Btw: the maximum number of characters for compatibility strings used here is 6. May I assume 6 is the maximum for TR_FUNC too?

PS: You meant "Switch/Event" rather than "Switch7Event", right?

mha1 commented:
@pfeerick  The switch selection sets means of activating/deactivating the SF. To be honest I'd leave it at Switch as this is really what is offered as selection options.

pfeerick commented:
> The switch selection sets means of activating/deactivating the SF. To be honest I'd leave it at Switch as this is really what is offered as selection options.

Well, yes and no... telem values aren't switches  :-P But I generally agree - I prefer switch, but if it were to change, I think of it more as the Trigger for an SF then Switch/Event... although that works also. btw, you might want to look at https://github.com/EdgeTX/edgetx/pull/3038 before you think about it as activating/deactivating to SF too much more ;)

> What is the maximum number of characters a string for x can have?

The answer to this one is *it depends* - there is no set length, but rather *"where is it used in the UI, and what fits that space"*... i.e. some places might only be three characters, others 7, others 11, etc. For the logical switch page and `STR_FUNC`... it looks like it could take up to 11 characters... i.e. "AND switch" fits in with a decent gap before the next column.

mha1 commented:
There you go, let's add a new word to the discussion: "Enable". Now we have "Switch", "Trigger", "Enable".

gui/128x64 uses STR_FUNC in model_logical_switches.cpp/menuModelLogicalSwitchOne(event_t event). I have no idea what the limit is here but suspect it is far less than 11 characters. Which radios to fall under 128x64?

mha1 commented:
Re: telemetry in Special Functions switch selection

I never noticed also due to the fact there is no telemetry icon on the switch selection list box. Bug?

![image](https://user-images.githubusercontent.com/5615068/213678711-b054ff3d-5941-47bc-b46b-26d213331aa7.png)

pfeerick commented:
> There you go, let's add a new word to the discussion: "Enable". Now we have "Switch", "Trigger", "Enable".

Nope, "Enable" is in the wheelhouse of that other PR, and will be a toggle switch, so no translation needed AFAIK. 

> Which radios to fall under 128x64?

Every single B&W radio except for the wide-screen ones :grin:

Remember I said "AND switch" fits? 😉 

![image](https://user-images.githubusercontent.com/5500713/213686315-3b2b89bc-a65e-4847-b550-3bf13550f7f8.png)

mha1 commented:
You beat me to it (obviously you own a faster PC). I fully agree, limit  looks to be 11 characters. So for TR_FUNXC everything up to 11 characters doesn't require to be made compatible.

![image](https://user-images.githubusercontent.com/5615068/213688832-7bf733da-774c-4bc5-a8d7-e51512cfb101.png)

HThuren commented:
> @HThuren
> 
> General question to TR_FUNC: I checked and found STR_FUNC being used in 3 places:
> 
> * gui/colorlcd Logical Switches
> * gui/colorlcd Special Functions
> * gui/128x64  Logical Swiitches
> 
> I guess the 128x64 case forces TR_FUNC to have a compatibility definition TR(x,y) if strings longer than n characters are used for colorlcd. What is the maximum number of characters a string for x can have? TR_FUNC has no compatibility definitions so far but has longer definitions for some languages, e.g, ES, FR, PT. That's why I believe those languages having longer strings already need to be changed to compatibility definitions depending on the answer to the above question about maximum number of characters in the compatibility case.
> 
> Switch definitions for Special Functions If there is consensus I can do this. Btw: the maximum number of characters for compatibility strings used here is 6. May I assume 6 is the maximum for TR_FUNC too?
> 
> PS: You meant "Switch/Event" rather than "Switch7Event", right?

after the further dialog, I think the right phase for SF_SWITCH are
**EN: #define TR_SF_SWITCH "Trigger"
DA: #define TR_SF_SWITCH "Udløser"**

mha1 commented:
Boy, this display variants thing is a can of worms. I was wondering why TR_FUNC was only used for 128x64 and colorlcd but not for 212x64 (all X9D variants). Turns out that there is a duplicate set of definitions for Logical Switches strings in all of the language files.  Here's the en.h portion:
```
#define TR_LSW_HEADERS_FUNCTION        "Function"
#define TR_LSW_HEADERS_V1              "V1"
#define TR_LSW_HEADERS_V2              "V2"
#define TR_LSW_HEADERS_ANDSW           "AND Switch"
#define TR_LSW_HEADERS_DURATION        "Duration"
#define TR_LSW_HEADERS_DELAY           "Delay"
```

I like the idea of having single sources rather than duplicate definitions. Should this be addressed? This is how imho it should be done:
```
#define TR_LSW_HEADERS_FUNCTION        TR_FUNC
#define TR_LSW_HEADERS_V1              TR_V1
#define TR_LSW_HEADERS_V2              TR_V2
#define TR_LSW_HEADERS_ANDSW           TR_AND_SWITCH
#define TR_LSW_HEADERS_DURATION        TR_DURATION
#define TR_LSW_HEADERS_DELAY           TR_DELAY
```

mha1 commented:
Do we have a decision on have a separate Switch string TR_SF_SWITCH for Special Functions? Go/no go?

robustini commented:
> Do we have a decision on have a separate Switch string TR_SF_SWITCH for Special Functions? Go/no go?

It would be more logical in my opinion.

mha1 commented:
@pfeerick 

Am I right in taking a thumbs up on single sourcing TR_LSW_HEADERS as go? And are we good with TR_SF_SWITCH?

Until yesterday I wasn't aware there was a renovation of the Special Functions layout going on. As we made some functional and textual changes to special_functions.h and .cpp with this PR I'd like to hit the pause button here until #2992 is merged into main. Then rebase this PR and migrate the functional and textual changes. Any idea when merging #2992 will take place?

ulfhedlund commented:
I',m a bit confused after reading all comments... 
What strings need to be translated?

ulfhedlund commented:
SE 
I hope this is what is required:
#define TR_PERIOD "Period"
#define TR_INTERVAL "Intervall"
#define TR_FUNC TR("Funk", "Funktion")
#define TR_SF_SWITCH "Trigger"

mha1 commented:
@ulfhedlund Perfect, thanks. I'll use `#define TR_FUNC  "Funktion"` instead of `#define TR_FUNC TR("Funk", "Funktion")` as the bw radios can use up to 11 characters. "Funktion"

pfeerick commented:
> Then rebase this PR and migrate the functional and textual changes. Any idea when merging #2992 will take place?

That would be great. Planning on review/merge early this week. I see no obvious problem with doing away with  `TR_LSW_HEADERS` ... ideally, if we can, replace those defines also if they are not actually needed.

mha1 commented:
@pfeerick I removed all TR_LSW_HEADERS_* defines in all language files and changed the TR_LSW_HEADERS in translations.h to use TR_* defines. Tested ok with X9D.

There are more single source opportunities in TR_PHASES_HEADERS, TR_LIMITS_HEADERS and TR_GVAR_HEADERS but only partial substitution are possible, e.g. using TR_NAME and TR_SWITCH.

ulfhedlund commented:
When going through the translations for logical switches I found the TR_LSW_DESCRIPTIONS. 
I can't see that it is used anywhere but if it is, or will be, someone need to verify that it is correct. I contains a duplicate string "Second variable or constant".

mha1 commented:
@ulfhedlund I couldn't find it either. Same goes for STR_GVAR_HEADERS. Looks like some code changes made them orphans. We could give it a try and delete those. The build runs should fall over if they are needed.

HThuren commented:
> @ulfhedlund I couldn't find it either. Same goes for STR_GVAR_HEADERS. Looks like some code changes made them orphans. We could give it a try and delete those. The build runs should fall over if they are needed.

Remember to build all model then, I myself discover diffences in different builds, or use "grep -r xx" :-)

mha1 commented:
I did a grep for STR_GVAR_HEADERS and TR_LSW_DESCRIPTIONS with no hit apart from translations.h/.cpp

ulfhedlund commented:
I just test built XT16S and Xlite firmware and they built fine without these strings. It is easy to locate unused or missing strings in the Companion translation files using the lupdate Qt tool but not so easy for the firmware translations. 

But perhaps this is better discussed elsewhere and not in this PR :)

mha1 commented:
@ulfhedlund You could set up a chore PR "cleanup translations".

ulfhedlund commented:
@mha1 : I would if I knew how to but I don't know how to make a chore PR I'm afraid.

I definitely see the need to go through the firmware translations though. Just found another possible duplication in TR_GLOBALVAR and TR_GLOBAL_VAR. There may be others as well, I suppose, but I don't now how to best locate them.

HThuren commented:
Thr Qt tool don't help with radio translations, only companion.

To locate, I simply compile them all, take a while, but then I'm sure not to break something I not was aware of.

ulfhedlund commented:
I agree, we shouldn't. That was not my intention either. 

However we now know there are some duplicates and unused strings in the firmware translations, so it may be worth going through them. If I can help I am willing to do so but I will leave it up to the (other) developers to decide how important this is.

mha1 commented:
@ulfhedlund I can offer to set up a PR and serve as editor for your content and discussion with the rest of the translators (@HThuren already volunteered to the builds and testing  :-)).

HThuren commented:
> @ulfhedlund I can offer to set up a PR and serve as editor for your content and discussion with the rest of the translators (@HThuren already volunteered to the builds and testing :-)).

ha ha, thats ok :-)

ulfhedlund commented:
OK great. Let's give it a go and see where it goes.

mha1 commented:
@ulfhedlund @HThuren Here's your PR https://github.com/EdgeTX/edgetx/pull/3082. Have at it!

pfeerick commented:
Just make the changes, do a single build locally to see if it passes, and then just push it to GitHub and let the automated builds do their job of testing the rest of the transmitter firmwares for you.

mha1 commented:
@pfeerick now that #2992 is merged: how do I do a proper rebase of this PR without messing everything up?

pfeerick commented:
Make sure your forks copy main is up to date, and then rebase off of main

i.e. Something like this
```
git checkout main
git fetch
git pull
git checkout DisplayLogPeriodZeroAsOFF
git rebase main
```

Now, you are going to have one conflict, but it looks to be straightforward to resolve that. Once you've done that, you'll need to force push the branch back to your repo. I've got a local copy I can push back up if it goes horribly wrong ;)

mha1 commented:
@philmoz SF page looks great.
@pfeerick I think migration was successful so I'd like to call it a day here. 
@ulfhedlund @HThuren Any further updates to translations should be done via PR https://github.com/EdgeTX/edgetx/pull/3082

Thanks for the great collaboration to Y'all

![SF](https://user-images.githubusercontent.com/5615068/214062392-16eb7515-9dfb-486e-8a37-d110fd37bea4.gif)

HThuren commented:
Hi @mha1, after - as agreed - compiled all language DA EN DE CZ SE IT CN TW PL JP,  all compile the radio part, but TW.

mha1 commented:
you mean tw is broken?

HThuren commented:
> you mean tw is broken?
right,
TR_SF_SWITCH and TR_PERIOD are missing in tw.h

mha1 commented:
Thanks, I'll fix it.

mha1 commented:
@HThuren I updated tw.h. Please try again.

HThuren commented:
> @HThuren I updated tw.h. Please try again.
I do it now

HThuren commented:
Well done, TW now compile for radio (and companion)
@zyren you maybe want to translate #define TR_PERIOD "Period" and #define TR_INTERVAL "Interval" in PR #3082

mha1 commented:
@HThuren Awesome, thank you very much for going through all languages. I think we covered all bases now.

HThuren commented:
@mha1 , try to update one of the files to restart the automatic check (only way I know of to do this)

pfeerick commented:
One minor change - Always use `STR_SF_SWITCH` for the sf switch text - i.e. it will always be Trigger regardless of the SF, unless the wording is changed later...

mha1 commented:
@pfeerick Thanks for merging this. To remove the logic to switch between STR_SWITCH and STR_SF_SWITCH entirely you can delete the line `StaticText *sfSwitchText = nullptr;` and change `sfSwitchText = new StaticText(line, rect_t{}, STR_SF_SWITCH, 0, COLOR_THEME_PRIMARY1);` to `new StaticText(line, rect_t{}, STR_SF_SWITCH, 0, COLOR_THEME_PRIMARY1);`

pfeerick commented:
Good point, thanks... I'll do that now before I forget as I don't think it will need to change dynamically just yet 😆

zyren commented:
**************************************************************************
CN
#define TR_PERIOD           "周期"
#define TR_INTERVAL         "间隔"
***************************************************************************
***************************************************************************
TW
#define TR_PERIOD           "週期"
#define TR_INTERVAL         "間隔"
**************************************************************************
